rs/sns/testing/src/pocket_ic.rs

```diff
@@ -124,3 +142,96 @@ async fn install_frontend_nns_canisters(pocket_ic: &PocketIc) {
     )
     .await;
 }
+
+// TODO @rvem: I don't like the fact that this struct definition is copy-pasted from 'canister/canister.rs'.
+// We should extract it into a separate crate and reuse in both canister and this crates.
+#[derive(CandidType)]
+pub struct TestCanisterInitArgs {
+    pub greeting: Option<String>,
+}
+
+pub async fn install_test_canister(pocket_ic: &PocketIc, args: TestCanisterInitArgs) -> CanisterId {
+    let topology = pocket_ic.topology().await;
+    let application_subnet_ids = topology.get_app_subnets();
+    let application_subnet_id = application_subnet_ids
+        .first()
+        .expect("No Application subnet found");
+    let features = &[];
+    let test_canister_wasm =
+        Wasm::from_location_specified_by_env_var("sns_testing_canister", features).unwrap();
+    install_canister_on_subnet(
+        pocket_ic,
+        *application_subnet_id,
+        Encode!(&args).unwrap(),
+        Some(test_canister_wasm),
+        vec![ROOT_CANISTER_ID.get()],
+    )
+    .await
+}
+
+pub async fn create_sns(
+    pocket_ic: &PocketIc,
+    dapp_canister_ids: Vec<CanisterId>,
+) -> (Sns, ProposalId) {
+    let sns_proposal_id = "1";
+    let create_service_nervous_system = CreateServiceNervousSystemBuilder::default()
+        .neurons_fund_participation(true)
+        .with_dapp_canisters(dapp_canister_ids)
+        .build();
+    let swap_parameters = create_service_nervous_system
+        .swap_parameters
+        .clone()
+        .unwrap();
+    assert_eq!(
+        swap_parameters.start_time, None,
+        "Expecting the swap start time to be None to start the swap immediately"
+    );
+    let (sns, proposal_id) =
+        propose_to_deploy_sns_and_wait(pocket_ic, create_service_nervous_system, sns_proposal_id)
+            .await;
+    await_swap_lifecycle(pocket_ic, sns.swap.canister_id, Lifecycle::Open)
```

Comment:
The issue seem to require this

> sns-testing scenario should close the swap using a sufficient amount of test participants with a sufficient amount of ICPs.

Do we know that it works like this in this `await_swap_lifecycle`? Or is it done somewhere else?

Reply:
> Or is it done somewhere else?

The swap is handled in `smoke_test_participate_and_finalize` slightly below. This helper considers required parameters in order to correctly close the swap